js/script.js

```diff
@@ -5,9 +5,39 @@ class Expenses {
   expName = document.querySelector("#exp-name");
 
   add() {
-    console.log("test add");
+    const priceVal = this.expPrice.value;
+    const nameVal = this.expName.value;
+  
+    if(!priceVal.length || !nameVal.length) {
+      this.expPrice.classList.add("warn-border");
+      this.expName.classList.add("warn-border");
+    } else {
+      const prevData = this.getData();
+      const obj = {name: nameVal, price: priceVal};
+  
+      if(!prevData) {
+        const arr = [];
+        arr.push(obj);
+        this.setData(arr);
+      } else {
+        prevData.push(obj);
+        this.setData(prevData);
+      }
+    }
+    this.expName.value = "";
+    this.expPrice.value = "";
   }
-
+  
+  getData() {
+    if(localStorage.expenses) return JSON.parse(localStorage.expenses);
+    return false;
```

Comment:
why return false if you can return an empty array? and there is no need to do the check above?